src/components/carousel/carousel.component.js

```diff
@@ -0,0 +1,366 @@
+  }
+
+  state = {
+    selectedSlideIndex: null, // Currently selected slide
+    disabled: false // Next/Previous buttons disabled state
+  };
+
+  /**
+   * A lifecycle method that is called after before initial render.
+   * Can set up state of component without causing a re-render
+   */
+  componentWillMount() {
+    const selectedIndex = Number(this.props.slideIndex) || Number(this.props.initialSlideIndex);
+    this.setState({ selectedSlideIndex: selectedIndex });
+  }
+
+  /** A lifecycle method that is called before re-render. */
+  componentWillReceiveProps(nextProps) {
+    if (typeof nextProps.slideIndex === 'undefined') return;
+
+    const newIndex = this.verifyNewIndex(nextProps.slideIndex);
+    const currentIndex = this.state.selectedSlideIndex;
+
+    if (newIndex === currentIndex) return;
+
+    if (newIndex > currentIndex) {
+      this.transitionDirection = NEXT;
+    } else {
+      this.transitionDirection = PREVIOUS;
+    }
+
+    this.handleSlideChange(newIndex);
+  }
+
+  /** Handles clicking on the previous button */
+  onPreviousClick() {
+    let newIndex = this.state.selectedSlideIndex - 1;
+    if (newIndex < 0) {
+      newIndex = this.numOfSlides() - 1;
+    }
+    this.transitionDirection = PREVIOUS;
+    this.handleSlideChange(newIndex);
+  }
+
+  /** Handles clicking on the next button */
+  onNextClick() {
+    const newIndex = (this.state.selectedSlideIndex + 1) % this.numOfSlides();
+    this.transitionDirection = NEXT;
+    this.handleSlideChange(newIndex);
+  }
+
+  /** Handles clicking slide selector */
+  onSlideSelection(ev) {
+    const newSlideSelection = Number(ev.target.value);
+    this.transitionDirection = newSlideSelection > this.state.selectedSlideIndex ? NEXT : PREVIOUS;
+    this.handleSlideChange(newSlideSelection);
+  }
+
+  /** Verifies the new index and corrects it if necessary */
+  verifyNewIndex(newIndex) {
+    if (newIndex < 0) {
+      // If the new index is negative, select the last slide
+      return this.numOfSlides() - 1;
+    }
+
+    if (newIndex > this.numOfSlides() - 1) {
+      // If the new index is bigger than the number of slides, select the first slide
+      return 0;
+    }
+
+    return newIndex;
+  }
+
+  /** Handle the slide change to the newIndex */
+  handleSlideChange(newIndex) {
+    if (isClassic(this.props.theme)) {
+      this.setState({ disabled: true, selectedSlideIndex: newIndex });
+    } else {
+      this.setState({ selectedSlideIndex: newIndex });
+    }
+
+    this.enableButtonsAfterTimeout();
+
+    if (this.props.onSlideChange) {
+      this.props.onSlideChange(newIndex, this.transitionDirection);
+    }
+  }
+
+  /** Re-enables the next and previous buttons after timeout */
+  enableButtonsAfterTimeout() {
+    if (isClassic(this.props.theme)) {
+      setTimeout(() => {
+        this.setState({ disabled: false });
+      }, TRANSITION_TIME);
+    }
+  }
+
+  /** Gets the props for the previous button */
+  previousButtonProps() {
```

Comment:
As the prop object has only one internal property you could change this to a one liner:
```
return !this.state.disabled ? { onclick: this.onPreviousClick } : {};
```